File: server/data/problems/121-best-time-to-buy-and-sell-stock/solutions/0-check-all-pairs/code.py
from typing import List

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        n = len(prices)
        
        # outer loop: the day we choose to buy
        for i in range(n):
            # inner loop: the day we choose to sell (must be after buy day)
            for j in range(i + 1, n):
                profit = prices[j] - prices[i]
                if profit > max_profit:
                    max_profit = profit
                    
        return max_profit

File: server/data/problems/121-best-time-to-buy-and-sell-stock/solutions/0-check-all-pairs/code.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;

        // outer loop: the day we choose to buy
        for (int i = 0; i < prices.length; i++) {
            // inner loop: the day we choose to sell (must be after buy day)
            for (int j = i + 1; j < prices.length; j++) {
                int currentProfit = prices[j] - prices[i];
                if (currentProfit > maxProfit) {
                    maxProfit = currentProfit;
                }
            }
        }
        
        return maxProfit;
    }
}